File: legacyjava/core/src/main/java/org/javafish/move/ScoredMove.java
package org.javafish.move;

import java.util.StringJoiner;

public record ScoredMove(Move move, short score) {

//    public Move getMove() {
//        return move;
//    }
//
//    public int getScore() {
//        return score;
//    }

    @Override
    public String toString() {
        return new StringJoiner(", ", ScoredMove.class.getSimpleName() + "[", "]")
                .add("move=" + move)
                .add("score=" + score)
                .toString();
    }

    public ScoredMove betterMove(Move move, short value) {
        return value > this.score ? move.scored(value) : this;
    }
}

File: ts-engine/src/ScoredMove.ts
import {Move} from './Move';

export class ScoredMove {
  readonly move: Move;
  readonly score: number;

  constructor(move: Move, score: number) {
    this.move = move;
    this.score = score;
  }

  // toString method
  toString(): string {
    return `ScoredMove[move=${this.move}, score=${this.score}]`;
  }
}
